### PR TITLE
refactor: add typescript [ECO-1017]

### PR DESCRIPTION
**Pull request by karantalapalli**
refactor: add typescript [ECO-1017]

### CODE REVIEW

File: src/components/HeaderContent.vue [hunk 1]
```diff
@@ -75,14 +86,14 @@ export default {
...
         jsonRtePath: ['notification_bar.announcement_text']
       });
-      let responsePages = await Stack.getEntries({
+      let responsePages: [PageResponse] = await Stack.getEntries({
         contentTypeUid: 'page'
       });
       let navHeaderList = response[0].navigation_menu;
       if (responsePages.length !== response.length) {
         responsePages.forEach(entry => {
           const hFound = response[0].navigation_menu.find(
```
Comment (surajcontentstack) [inline src/components/HeaderContent.vue:95]: use `navHeaderList` variable

File: src/views/BlogContent.vue [hunk 2]
```diff
@@ -49,22 +49,37 @@
   <Skeletor v-else height="100vh" />
 </template>
 
-<script>
+<script lang="ts">
+
+interface List {
+    author: [];
```
Comment (surajcontentstack) [inline src/views/BlogContent.vue:55]: type can be `Array<any>`

File: src/views/BlogContent.vue [hunk 3]
```diff
@@ -74,13 +89,13 @@ export default {
   },
   methods: {
     async getData() {
-      const archived = [];
-      const recentPost = [];
+      const archived = [] as any;
```
Comment (surajcontentstack) [inline src/views/BlogContent.vue:92]: as `Array<any>`

File: src/views/BlogContent.vue [hunk 4]
```diff
@@ -97,11 +112,10 @@ export default {
       this.archivedList = archived;
       this.$store.dispatch('setPage', data[0]);
       this.$store.dispatch('setBlogpost', list);
-      document.title = this.banner.title;
-      const element = document.getElementsByClassName('cslp-tooltip');
+      const element: any = document.getElementsByClassName('cslp-tooltip');
```
Comment (surajcontentstack) [inline src/views/BlogContent.vue:115]: check if you can use `this$refs` to access DOM elements

**Review by surajcontentstack — CHANGES_REQUESTED** [review]

File: src/components/AboutSectionBucket.vue [hunk 5]
```diff
@@ -34,8 +34,26 @@
...
+
+interface Buckets{
+  title_h3: string;
+  description: string;
+}
+
+interface Data{
+  title_h2: string;
+  bucket: Buckets;
+}
```
Comment (abhishek305) [inline src/components/AboutSectionBucket.vue:47]: For individual components have a common file from were you export **interface** using models
reference: https://dev.to/jwp/typescript-models-and-javascript-objects-31b3

File: src/components/BlogSection.vue [hunk 6]
```diff
@@ -37,8 +37,33 @@
...
+interface Blogs {
+  featured_image: Image;
+  url: string;
+}
+
+interface Data {
+  title_h2: string;
+  view_articles: Article;
+  featured_blogs: Blogs;
+}
```
Comment (abhishek305) [inline src/components/BlogSection.vue:56]: For individual components have a common file from were you export interface using models.

File: src/components/Devtools.vue [hunk 7]
```diff
@@ -44,9 +44,37 @@
...
+  title: string;
+}
+
+interface Response {
+    page?: string;
+    blog_post?: string;
+    headers: Header;
+    footer: Footer;
+}
+
```
Comment (abhishek305) [inline src/components/Devtools.vue:73]: use models as required

File: src/components/FooterContent.vue [hunk 8]
```diff
@@ -46,12 +46,23 @@
...
+
+interface PageResponse {
+  title: string;
+  url: string;
+}
+
+interface Links {
+  title: string;
+}
+
```
Comment (abhishek305) [inline src/components/FooterContent.vue:59]: use models

**Review by abhishek305 — COMMENTED** [review]
use models instead of interfaces wherever applicable @karantalapalli

**Review by surajcontentstack — APPROVED** [review]

**Review by abhishek305 — APPROVED** [review]

**Pull request merged**